compiler/noirc_driver/src/lib.rs

```diff
@@ -174,6 +174,11 @@ pub struct CompileOptions {
     #[arg(long, default_value = "false")]
     pub pedantic_solving: bool,
 
+    /// Accept a single '.nr' file through STDIN instead of reading from the
+    /// root directory
+    #[arg(long, hide = true)]
+    pub debug_compile_stdin: bool,
```

Comment:
Could you expand the comment here `debug_compile_stdin`. This is where I would look if I was unsure of why we call `with_maybe_dummy_workspace` for the compile command.